### PR TITLE
Add better error handling

### PR DESCRIPTION
**Pull request by sagikazarmark**
## Overview
Add better error handling

4xx error logging is removed: we can enable access logs for that
Error handler logs 5xx error (when necessary)

It's going to become significantly easier, once we move away from HTTP handlers for core logic (most error mapping can happen outside of controllers)

### CODE REVIEW

File: internal/server/router/meter.go [hunk 1]
```diff
@@ -3,23 +3,27 @@ package router
...
+	ctx := contextx.WithAttr(r.Context(), "operation", "listMeters")
+
 	namespace := a.config.NamespaceManager.GetDefaultNamespace()
 
 	meters, err := a.config.Meters.ListMeters(r.Context(), namespace)
 	if err != nil {
 		err := fmt.Errorf("list meters: %w", err)
-		models.NewStatusProblem(r.Context(), err, http.StatusInternalServerError).Respond(logger, w, r)
+
+		a.config.ErrorHandler.HandleContext(ctx, err)
```
Comment (hekike) [inline internal/server/router/meter.go:24]: this is such a commonly used function, wondering if we could add it to router and just be `a.HandleError(ctx, err))` or bake it it into NewStatusProblem.Respond

File: internal/server/router/meter_query.go [hunk 2]
```diff
@@ -12,33 +13,42 @@ import (
...
+	"github.com/openmeterio/openmeter/pkg/contextx"
 	"github.com/openmeterio/openmeter/pkg/models"
 )
 
 // QueryMeter queries the values stored for a meter.
 func (a *Router) QueryMeter(w http.ResponseWriter, r *http.Request, meterIDOrSlug string, params api.QueryMeterParams) {
-	logger := slog.With("operation", "queryMeter", "id", meterIDOrSlug, "params", params)
+	ctx := contextx.WithAttr(r.Context(), "operation", "queryMeter")
+	ctx = contextx.WithAttr(ctx, "id", meterIDOrSlug)
+	ctx = contextx.WithAttr(ctx, "params", params) // TODO: HOW ABOUT NO????
```
[the remark this answers is not in the document]
Comment (sagikazarmark) [reply internal/server/router/meter_query.go:24]: Yeah, yeah, that was born out of my frustration.

File: internal/server/router/meter.go (hunk 1 above)
Comment (sagikazarmark) [reply internal/server/router/meter.go:24]: If we add anything back to the status problem, I'm gonna have to cry myself to sleep. 😄 

The right solution is not baking business logic into HTTP handlers. Then we don't have to handle and map every error manually every single time, but we can rely on middleware instead.

Adding logging to the status problem was always a half measure (if you can call it that). If it doesn't hurt enough though, we are never going to fix it.

tl;dr The explicitness will lead us to a better solution eventually.

**Review by hekike — APPROVED** [review]

**Review by tothandras — APPROVED** [review]
